From 3f564a5a4e8ceb04d7349fdffd2354134f211601 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Wed, 15 Feb 2023 11:26:23 +0530
Subject: [PATCH 040/120] usb: hcd: Call set_mode for the PHY

add support to call set_mode callback of all the PHYs
linked connected to ehci/ohci generic ctrls

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 16407f70851db0f277872d8039538cff049f5869 usb: hcd: Call set_mode for the PHY]
Change-Id: I88d9ebe7657459235e801fd144c1f296e9f13062
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/303895
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 drivers/usb/host/ehci-generic.c | 6 ++++++
 drivers/usb/host/ohci-generic.c | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/usb/host/ehci-generic.c b/drivers/usb/host/ehci-generic.c
index 1ae3619ce25..a23c8e3211d 100644
--- a/drivers/usb/host/ehci-generic.c
+++ b/drivers/usb/host/ehci-generic.c
@@ -98,6 +98,12 @@ static int ehci_usb_probe(struct udevice *dev)
 	if (err)
 		goto regulator_err;
 
+	err = generic_phy_set_mode(&priv->phy, PHY_MODE_USB_HOST, 0);
+	if (err) {
+		dev_dbg(dev, "failed to set mode on usb phy\n");
+		goto phy_err;
+	}
+
 	hccr = map_physmem(dev_read_addr(dev), 0x100, MAP_NOCACHE);
 	hcor = (struct ehci_hcor *)((uintptr_t)hccr +
 				    HC_LENGTH(ehci_readl(&hccr->cr_capbase)));
diff --git a/drivers/usb/host/ohci-generic.c b/drivers/usb/host/ohci-generic.c
index cc44226f5e0..1dd659130e1 100644
--- a/drivers/usb/host/ohci-generic.c
+++ b/drivers/usb/host/ohci-generic.c
@@ -54,6 +54,10 @@ static int ohci_usb_probe(struct udevice *dev)
 	if (err)
 		goto reset_err;
 
+	err = generic_phy_set_mode(&priv->phy, PHY_MODE_USB_HOST, 0);
+	if (err)
+		goto phy_err;
+
 	err = ohci_register(dev, regs);
 	if (err)
 		goto phy_err;
-- 
2.47.2

